src/service/database/index.ts

```diff
@@ -46,9 +48,54 @@ export class DatabaseService {
 
     for (const row of data) {
       const [name, description] = row as string[];
-      databases.push(new DatabaseModel({ name, description }));
+      databases.push(new DatabaseModel({ name, description }, this.context));
     }
 
     return databases;
   }
+
+  async create(
+    name: string,
+    options: CreateDatabaseOptions = {}
+  ): Promise<DatabaseModel> {
+    if (options.fail_if_exists == undefined) {
+      options.fail_if_exists = true;
+    }
+    let query: string =
+      `CREATE DATABASE ${options.fail_if_exists ? "" : "IF NOT EXISTS "} "${name}"`;
+
+    const queryParameters = [];
+    let attachedEnginesSql: string = ""
+    if (
+      Array.isArray(options.attached_engines) &&
+      options.attached_engines.length > 0
+    ) {
+      if (options.attached_engines[0] instanceof EngineModel) {
```

Comment:
this looks like a duplicate. And javascript can have values in array of any type. Can we have something like this instead: 
```typescript

const attachedEngines = options.attached_engines.map(engine => engine instanceof EngineModel ? `"${engine.name}"` : `"${engine}"`).join(" ");
attachedEnginesSql = ` ATTACHED_ENGINES = (${attachedEngines})`;
```